Mjml.Net/Components/Body/ButtonComponent.cs

```diff
@@ -0,0 +1,170 @@
+                .Attr("valign", node.GetAttribute("vertical-align"))
+                .Style("border", node.GetAttribute("border"))
+                .Style("border-bottom", node.GetAttribute("border-bottom"))
+                .Style("border-left", node.GetAttribute("border-left"))
+                .Style("border-right", node.GetAttribute("border-right"))
+                .Style("border-top", node.GetAttribute("border-top"))
+                .Style("border-radius", borderRadius)
+                .Style("cursor", "auto")
+                .Style("font-style", fontStyle)
+                .Style("height", height)
+                .Style("mso-padding-alt", innerPadding)
+                .Style("text-align", textAlign)
+                .Style("background", backgroundColor);
+
+            renderer.ElementStart(buttonHtmlTag)
+                .Attr("href", node.GetAttribute("href"))
+                .Attr("rel", node.GetAttribute("rel"))
+                .Attr("name", node.GetAttribute("name"))
+                .Attr("target", !string.IsNullOrEmpty(href) ? node.GetAttribute("target") : null)
+                .Style("display", "inline-block")
+                .Style("width", CalculateButtonWidth(node))
+                .Style("background", backgroundColor)
+                .Style("color", node.GetAttribute("color"))
+                .Style("font-family", node.GetAttribute("font-family"))
+                .Style("font-style", fontStyle)
+                .Style("font-size", node.GetAttribute("font-size"))
+                .Style("font-weight", node.GetAttribute("font-weight"))
+                .Style("line-height", node.GetAttribute("line-height"))
+                .Style("letter-spacing", node.GetAttribute("letter-spacing"))
+                .Style("margin", "0")
+                .Style("text-decoration", node.GetAttribute("text-decoration"))
+                .Style("text-transform", node.GetAttribute("text-transform"))
+                .Style("letter-spacing", node.GetAttribute("letter-spacing"))
+                .Style("padding", innerPadding)
+                .Style("mso-padding-alt", "0px")
+                .Style("text-align", textAlign)
+                .Style("border-radius", borderRadius);
+
+            renderer.Content(node.GetContent());
+
+            renderer.ElementEnd(buttonHtmlTag);
+            renderer.ElementEnd("td");
+            renderer.ElementEnd("tr");
+            renderer.ElementEnd("table");
+        }
+
+        private static string? CalculateButtonWidth(INode node)
+        {
+            string? width = node.GetAttribute("width");
+
+            if (string.IsNullOrEmpty(width) || !width.Contains("px", StringComparison.OrdinalIgnoreCase))
+            {
+                return null;
+            }
+
+            if (!double.TryParse(width?.Replace("px", string.Empty, StringComparison.InvariantCultureIgnoreCase), NumberStyles.Integer, CultureInfo.InvariantCulture, out double widthParsed))
```

Comment:
I have an extension method for that which is allocation free:

https://github.com/SebastianStehle/mjml-test/blob/main/Mjml.Net/Components/NodeExtensions.cs#L132

Reply:
Nice, I'll switch to that.

Reply:
It's probably because it's getting late but I'm unable to access this from `node.ParseNumber(width)`. 

```
Error	CS1061	'INode' does not contain a definition for 'ParseNumber' and no accessible extension method 'ParseNumber' accepting a first argument of type 'INode' could be found (are you missing a using directive or an assembly reference?)
```

Reply:
Oh my gosh. It's definitely getting late! I should be using `GetAttributeNumber()` 🤦‍♂️